src/java.base/share/classes/java/lang/String.java

```diff
@@ -2987,7 +2987,7 @@ public String concat(String str) {
         if (str.isEmpty()) {
             return this;
         }
-        return StringConcatHelper.simpleConcat(this, str);
+        return StringConcatHelper.doConcat(this, str == null ? "null" : str);
```

Comment:
```suggestion
        if (isEmpty()) {
            return str;
        }
        return StringConcatHelper.doConcat(this, str);
```
`str.isEmpty()` already does the null check.